lib/openscap_parser/value.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+module OpenscapParser
+  class Value < XmlNode
+    include OpenscapParser::Util
+
+    def id
+      @id ||= parsed_xml['id']
+    end
+
+    def description
+      @description ||= newline_to_whitespace(parsed_xml.at_css('description')&.text)
+    end
+
+    def title
+      @title ||= parsed_xml.at_css('title')&.text
+    end
+
+    def type
+      @type ||= parsed_xml['type'] || 'string'
+    end
+
+    def lower_bound(selector = nil)
+      @lower_bounds ||= parsed_xml.xpath('lower-bound').each_with_object({}) do |element, lb|
+        lb[element.at_xpath('@selector')&.text] = element&.text
+      end
+      selector ? @lower_bounds[selector] : @lower_bounds[nil] || @lower_bounds[""]
+    end
+
+    def upper_bound(selector = nil)
+      @upper_bounds ||= parsed_xml.xpath('upper-bound').each_with_object({}) do |element, ub|
+        ub[element.at_xpath('@selector')&.text] = element&.text
+      end
+      selector ? @upper_bounds[selector] : @upper_bounds[nil] || @upper_bounds[""]
+    end
+
+    def value(selector = nil)
+      @values ||= parsed_xml.xpath("value").each_with_object({}) do |element, rv|
+        rv[element.at_xpath('@selector')&.text] = element&.text
+      end
+      selector ? @values[selector] : @values[nil] || @values[""] || parsed_xml.xpath("value")[0]&.text
```

Comment:
This is way too complex for a one-liner, also the `""` is a red flag.

Reply:
Yeah, I figured it would be too much.. I'll split it up
But the default lower-bound/upper-bound will be the element with no `selector` attribute or the element where `selector=""` so that's why I'm checking `@values[""]`

Reply:
I think the `nil` and `""` should be joined together. Also the last one can be simply `@values.values.first` as I said on the chat. Instead of the ternary + OR logic please use an early return in case `selector` is set:
```
return @values[selector] if selector

@values[nil] || @values.values.first
```

Reply:
You can ensure on line 38 that the key always becomes `nil`, even if the selector is an empty string.